integration/docker/Dockerfile-dev

```diff
@@ -9,86 +9,24 @@
-# ADD then chown in two steps will double the image size
-#   See - https://stackoverflow.com/questions/30085621/why-does-chown-increase-size-of-docker-image
-#       - https://github.com/moby/moby/issues/5505
-#       - https://github.com/moby/moby/issues/6119
-# ADD with chown doesn't chown the files inside tarball
-#   See - https://github.com/moby/moby/issues/35525
-FROM alpine:3.10.2 AS alluxio-extractor
-# Note that downloads for *-SNAPSHOT tarballs are not available.
-ARG ALLUXIO_TARBALL=http://downloads.alluxio.io/downloads/files/2.9.0-SNAPSHOT/alluxio-2.9.0-SNAPSHOT-bin.tar.gz
-# (Alert):It's not recommended to set this Argument to true, unless you know exactly what you are doing
-ARG ENABLE_DYNAMIC_USER=false
-
-ADD ${ALLUXIO_TARBALL} /opt/
-# Remote tarball needs to be untarred. Local tarball is untarred automatically.
-# Use ln -s instead of mv to avoid issues with Centos (see https://github.com/moby/moby/issues/27358)
-RUN cd /opt && \
-    (if ls | grep -q ".tar.gz"; then tar -xzf *.tar.gz && rm *.tar.gz; fi) && \
-    ln -s alluxio-* alluxio
-
-RUN if [ ${ENABLE_DYNAMIC_USER} = "true" ] ; then \
-       chmod -R 777 /opt/* ; \
-    fi
-
-# Configure JAVA_HOME
-FROM centos:7 as build_java8
-ENV JAVA_HOME /usr/lib/jvm/java-1.8.0-openjdk
-
-FROM centos:7 as build_java11
-ENV JAVA_HOME /usr/lib/jvm/java-11-openjdk
-
-FROM build_java${JAVA_VERSION} AS final
-
-ADD dockerfile-common.sh /
-
-# Install libfuse2 and libfuse3. Libfuse2 setup is modified from cheyang/fuse2:ubuntu1604-customize to be applied on centOS
-WORKDIR /
-
-ENV MAX_IDLE_THREADS "64"
+USER root
 
 RUN \
     yum update -y && yum upgrade -y && \
-    yum install -y ca-certificates pkgconfig wget udev git && \
-    yum install -y gcc gcc-c++ make cmake gettext-devel libtool autoconf && \
-    yum clean all && \
-    ./dockerfile-common.sh install-libfuse2 && \
-    yum install -y fuse3 fuse3-devel fuse3-lib
-
-ENV LD_LIBRARY_PATH "/usr/local/lib:${LD_LIBRARY_PATH}"
-
-ARG ALLUXIO_USERNAME=alluxio
-ARG ALLUXIO_GROUP=alluxio
-ARG ALLUXIO_UID=1000
-ARG ALLUXIO_GID=1000
-ARG ENABLE_DYNAMIC_USER=true
-
-# Add Tini for alluxio helm charts (https://github.com/Alluxio/alluxio/pull/12233)
-# - https://github.com/krallin/tini
-ENV TINI_VERSION v0.18.0
-ADD https://github.com/krallin/tini/releases/download/${TINI_VERSION}/tini /usr/local/bin/tini
-RUN chmod +x /usr/local/bin/tini
-
-RUN \
-    yum install -y java-1.8.0-openjdk-devel java-1.8.0-openjdk && \
     yum install -y java-11-openjdk-devel java-11-openjdk && \
-    yum install -y unzip vim && \
+    yum install -y ca-certificates pkgconfig wget udev git gcc gcc-c++ make cmake gettext-devel libtool autoconf unzip vim && \
     yum clean all
 
+# Create a symlink for setting JAVA_HOME depending on java version. ENV cannot be set conditionally.
+RUN ln -s /usr/lib/jvm/java-1.8.0-openjdk /usr/lib/jvm/java-8-openjdk
+ARG JAVA_VERSION 8
+ENV JAVA_HOME /usr/lib/jvm/java-${JAVA_VERSION}-openjdk
```

Comment:
Not a requirement for this PR: 
can we set java version, install java package and set java home based on the input argument in both Dockerfile and Dockerfile-dev?
so that these codes are not needed to be added

Reply:
Let's discuss this offline. I don't completely get what you mean here